packages/rollup-core/src/app/data/producers/l2-chain-data-persister.ts

```diff
@@ -140,6 +144,23 @@ export class L2ChainDataPersister extends ChainDataProcessor {
       )}${response.v.toString(16)}`
     }
 
+    log.debug(
+      `L2 Tx Output for block ${receipt.blockNumber}: ${JSON.stringify(res)}`
+    )
+
     return res
   }
+
+  private static parseBigNumber(data: any): BigNumber {
+    if (!data) {
+      return undefined
+    }
+    if (typeof data === 'string') {
+      return new BigNumber(remove0x(data))
+    }
+    if (typeof data.toHexString === 'function') {
+      return new BigNumber(data.toHexString())
+    }
+    return new BigNumber(data.toString('hex'))
```

Comment:
Same radix comment here and line 162.  Also, I think this being here makes the case where `typeof data === 'string'` redundant.

Reply:
This one should handle `BigNum`s (not the same as `BigNumber` 😢) and Buffers. I think adding a `remove0x(...)` around the `toString(...)` would work for strings as well, but will probably keep the `typeof data === string` case separate, as it _should_ be a string, and consolidating them will push the expected case down to the catch-all adding an unnecessary if, call to `toString(...)`, and call to `remove0x(...)`